### ISSUE TITLE
uBlock Origin on Firefox Preview

### ISSUE THREAD

**Issue by gorhill** [uBlockOrigin/uBlock-issues#1027]
This issue contains the email exchange I had with Mozilla's @brampitoyo regarding the re-design of uBO's user interface, originally aimed at Firefox Preview but which is currently used on all platforms as of uBO version 1.27.0.

The email exchange is reproduced here with @brampitoyo's permission. Note that I was originally contacted by Mozilla's @kewisch regarding the project of re-designing uBO's user interface, who put me in contact with @brampitoyo.

I am very pleased with the new user interface and needless to say this re-design is just not something I would have been able to come up on my own, @brampitoyo was essential to make this happens.

Not all suggested changes have been yet addressed in the user interface, so this issue is to be kept opened and further discussion regarding user interface improvements can continue here.

**Comment by gorhill** [comment]
Bram Pitoyo
2019-11-12

-----

Hi Raymond,

So glad to hear from you, and to have a chance to work together.

> Maybe some buttons are not really needed on mobile?


Agreed. Generally, the more buttons we can take out, the less touch targets we’ll have per page. I’d be happy to assist you in doing an audit of features that are okay to excise for mobile, but I must first confess that I’m not a domain expert!


> I wonder whether the text will neatly fits for all languages such that the layout stays the same in all languages.

 
Our text list items can scale up to 2–3–4 lines. Hopefully this can help rectify the language-fitting issue.

For icon + label items, the space is more limited, so they’re not as ‘stretchable’. I’d advise limiting the amount of items in this category. But a good alternative is to shorten some strings.

For example, instead of saying “Open the logger”, we may be able to say “Logger”. Instead of saying “Enter element zapper mode”, we may be able to say “Element zapper”. I’m not sure if this will impact clarity, though!


> How does the overview panel fits in there?


I spent most of today taking a deep dive into features and sub-pages within uBlock Origin. I was able to adapt them and address some of your thoughts around information density.

The good news is, I found that these adaptations can, for the most part, be mostly visual. This means that no new feature or change in functionality is required. In some cases, that’s not possible. Have a browse below.


### Main user interface

For starters, I thought that the entry point for the Overview panel should be clearly labelled.

<img width="360" src="https://user-images.githubusercontent.com/585534/81928693-ad748280-95b3-11ea-94fa-e9478de0fbb7.png">

It’s up to you whether this panel is expanded downwards and upwards, or whether it opens as a sub-page. In the example above, it opens in a sub-page.


### Overview panel

When users tap “Overview panel”, a familiar-looking table appears.


<img width="360" src="https://user-images.githubusercontent.com/585534/81928772-d3018c00-95b3-11ea-8272-e973bce0dc98.png">

The Overview panel has a high information density, so adapting it was challenging. My main concern was to have sufficiently large tap targets (48dp) while not losing any feature or complexity.

In the example above, Global and Local Rules are modes that you switch back and forth from/to. Allow/block controls are presented as status to the right of each domain name, and can be modified via a dropdown menu (if “Advanced mode”) is switched on.


### Domain details

One thing that was missing from the Overview panel is +++ ---. I want not only to preserve them, but give them ample space so you can explain as much as you’re able to.

To view this domain information, the user can tap the domain name. It will open another sub-page. It has enough space to contain almost anything you want.

<img width="360" src="https://user-images.githubusercontent.com/585534/81928857-f4627800-95b3-11ea-8478-c912077bf7bd.png">


### Dashboard

The dashboard contains a lot of tabs. We have a scrollable tab bar design that can fit all of them.

<img width="360" src="https://user-images.githubusercontent.com/585534/81928923-0e9c5600-95b4-11ea-826f-294f33f7724d.png">

These examples may not sufficiently capture the full complexity of uBlock Origin, but hopefully they can still give you a taste of the flexibility that our Android layouts can offer.

Let me know what you think of these? If it’s helpful, I’d be happy to revise these designs and/or continue adapting more pages.

**Comment by gorhill** [comment]
Raymond Hill
2019-11-21

-----

Hello Bram,

Sorry for the delayed response. Regarding:

> The Overview panel has a high information density, so adapting it was challenging. My main concern was to have sufficiently large tap targets (48dp) while not losing any feature or complexity.

I consider the high information density to be a key usability quality for advanced users (which is the purpose of that panel), as it allows to see a lot *at a glance*:

- See what got blocked/allowed on which domains, along with a sense of quantity of the number of blocked/allowed requests (the --/++).

- See all the rules currently in effect, and whether a local rule in inherited from a global one (i.e. pale shade versus dark one)

As an advanced user myself, I consider the "at a glance" aspect to be a key feature and would be unhappy to lose this when I have to assess/modify ruleset for the current site, so I consider this is something that must be as close to what it currently is.

**Comment by gorhill** [comment]
Raymond Hill
2019-11-30

-----

Hi Bram, feedback regarding the abstract.

I don't think the overview panel should expand before the extra tools,
which are part of the basic panel. It should appear below the basic
panel -- it's not always to be used for interactivity purpose but also
just in read-only mode. The extra tools may affect the length of the
overview panel list, the list may change as more requests are made by
the page, so it's better that it does not keep pushing further down
the extra tools. This is how it currently works in the mobile version
of uBO.

The horizontal space available for the hostnames is too small. The
abstract sceenshot show a case where all hostnames fit but in the real
world this is going to be an issue. We need more horizontal space for
the hostnames. Separating the cells with a single pixel space will
also help recover space -- this is why this is so in the current
version of uBO.

Also a detail, the minus sign should be `\u2212` so that it's the same
width as the plus sign.

The toggle for JavaScript in the basic panel really need to be visible
by default, we may need to remove "Block all pop-ups" toggle.

**Comment by gorhill** [comment]
Bram Pitoyo
2019-12-01

-----

Hi Raymond, thanks for your feedback. I’m addressing them below.

> …the overview panel […] should appear below the basic panel […] This is how it currently works in the mobile version of uBO.

When I used uBlock Origin on Firefox for Android (Fennec), the overview panel opens to the side. But your suggestion of opening below the basic panel makes a lot of sense. See below:

<img width="360" src="https://user-images.githubusercontent.com/585534/81929327-d6e1de00-95b4-11ea-9242-52c1a4369521.png">

You’ll also notice that I’ve removed the “Block all pop-ups” toggle and show “Disable JavaScript” by default. I’ve also switched icons from our house style to FontAwesome – the same ones you’ve used on uBO – to present a more accurate picture.

My question: does it still make sense to call this area “Overview” and hide its content by default? (When hidden, the downwards arrow icon faces up).

> The horizontal space available for the hostnames is too small. […] We need more horizontal space for the hostnames. Separating the cells with a single pixel space will also help recover space -- this is why this is so in the current version of uBO.

I took your suggestion and applied it. I’ve also switched to the correct minus sign (`\u2212`). See below:

<img width="360" src="https://user-images.githubusercontent.com/585534/81929424-fa0c8d80-95b4-11ea-97dc-836cfadd5445.png">

You’ll note that the hostname font size is now smaller – 14px → 12px – which may be as small as the font size on desktop. Hopefully this will make long URLs fit better.

One thing I didn’t change was the height of each row – it’s still 24px – and this is to keep the target large enough to tap. This, assuming that some mobile uBO users want to modify the Overview panels? If modification is less likely, we can narrow the row height down to 20px, 16px and so on.

Should you like to inspect the pixel-level design, the link stays the same:
https://share.goabstract.com/54f1465e-8d1c-4e7b-9ae4-f37e21bef7fe

What do you think? Thanks a lot for your feedback!

**Comment by gorhill** [comment]
Raymond Hill
2019-12-05

-----

Regarding:

> When I used uBlock Origin on Firefox for Android (Fennec), the overview panel opens to the side.

Yes, uBO will by default opens the overview panel on the side. However it will calculate whether the panel fits well enough in the current viewport and if not it will send the panel at the bottom. This is why you can still get the panel on the side in landscape mode on a mobile display, or at the bottom on Firefox desktop when we pin uBO in the overflow menu.

So anyways, I think at this point we should start implementing the changes so that we can see and feel the real thing? How is this going to work? Do I make the changes on my side or are you going to make them on your side? The way I see it is to create alternate popup.html/css files and to enable them using browserAction.setPopup().

**Comment by gorhill** [comment]
Bram Pitoyo
2019-12-08

-----

> I think at this point we should start implementing the changes so that we can see and feel the real thing? How is this going to work? Do I make the changes on my side or are you going to make them on your side?

Hi Raymond, I agree. I think that you should make changes on your side.

As far as implementation goes, it’s up to you whether there should be an alternate set of popup.html/css files, or whether there’s only one responsive stylesheet.

When you post some screenshots of your process, I’d be able to comment and send along my feedback.

In the near-term, this responsive design would fit well for purpose.

In the future, we hope to have some sort of a Firefox Mobile web component library for all add-on developers to use. You can already use Google’s Material Web Components for this purpose today, but it may not cover unique challenges that add-ons pose.

Talk soon,
Bram

**Comment by gorhill** [comment]
Bram Pitoyo
2019-12-15

-----

Hi Raymond (and cc Philipp Kewisch),

If you’ve got screenshots or stylesheets, I‘d be happy to have a look and/or assist this week – just let me know how I can help!

For date and other technical questions, Philipp or Emily will assist.

At the end of my work week this week, I am planning to take some time off for Christmas and New Years, to return in late January.

–Bram

**Comment by gorhill** [comment]
Raymond Hill
2019-12-16

-----

Sorry I haven't had much time to work on uBO recently so I haven't started to work on this yet.

**Comment by gorhill** [comment]
Raymond Hill
2020-01-25

-----

Hi Bram,

I've committed a first draft of the redesigned popup panel, it's in the
dev build 1.24.3b11 of uBO, available at:

https://github.com/gorhill/uBlock/releases

The new popup panel must be activated though as follow:

- Go to uBO's "Settings" pane in uBO's dashboard
- Check "I am an advanced user" option
    - This is necessary to gain access to "advanced settings"
- Click the gear icon which appear after "I am an advanced user" once the option is checked
- A page opens with various text editor-based settings
- There should be an entry named "uiFlavor", which is set to "unset" by default
- Change the value to "fenix" and click "Apply changes"
- Now the next time you restart uBO, the redesigned popup panel will be used by uBO

So regarding the redesigned popup panel, I faithfully followed your design as per previous email. However I also made these fine tuning:

- I added a separator line below the power button
- I renamed "Overview" to "More", as in reality the revealed panel is really to gain access to more detailed information and/or advanced controls.
- The refresh, padlock and eraser buttons were not part of your redesigned, so for now I have relocated them at the top along the power button
- I made the power button sticky such that scrolling down what could be a long list of hostnames in the "More" panel will keep the power button and adjacent refresh/padlock/eraser buttons in view

So I guess we can consider this a first draft, I am open to change whatever you think is worth changing further, while for the "at a glance" firewall panel I prefer to keep it as is as I think it's difficult to improve upon (though font size, dimensions can probably be fine-tuned) as all the space available is used.

For the icons, they are currently all from forkawesome set of icons[1], but if you think better icons can be used then I am also open to improve this part -- since it's a redesigned, might as well maybe revisit everything worth to revisit.

**Comment by gorhill** [comment]
Bram Pitoyo
2020-01-25

-----

Hi Raymond,

Thanks heaps – very much – for your very prompt reply about the badge text and now the redesigned popup panel. We can’t wait to test it, and really hope that it will look great in Firefox Preview!

We will all be in Berlin for a Mozilla all hands even next week. Our replies might come late, unfortunately, but we’ll reply with feedback as soon as we can.

Until then,
Bram

**Comment by gorhill** [comment]
Bram Pitoyo
2020-02-05

-----

Hi Raymond,

I was going to test the uiFlavour change that you’ve made, but unfortunately dev build 1.24.3b11 is no longer available to download.

When I downloaded the latest dev build 1.24.5b1, I wasn’t able to find uiFlavour under advanced settings. And when I tried adding uiFlavour manually, it didn’t work.

Do I need to download another special build to see the UI changes you’ve made? (If it’s too much work for you to make a special build, then it’s not a big deal – I’m happy to see a few screenshots from you).

Thanks again for following the design I sent out. I hope you felt like it was a collaborative process, and that uBO now looks better on mobile.

For the moment, I wonder whether you’ve looked at our icon library?
https://design.firefox.com/icons/viewer/#

At the moment, our Android selection is limited, but we have heaps more in this folder, which is ready for you to use.

https://drive.google.com/drive/folders/1_DeNm4b9z94YhSnxQwhLu6f4mgghf9DN?usp=sharing

There are a couple of icons we have, which looks similar to ones you use. I’ll attach them on this email to save you time and effort:

- Energy.svg – similar to element zapper
- List.svg – similar to logger
- Preference.svg – not similar to dashboard, but we also use it for settings/preferences
- Pop-ups.svg – similar to block pop-ups
- Readermode Size.svg – similar to block remote fonts

Unfortunately, we don’t yet have the full suite of icon for you to do a total replacement. And I know you want interface elements to look like they belong together. Have you considered Font Awesome 5?

When I see the new interface, I’ll come back with feedback ASAP.

Thanks again,
Bram

**Comment by gorhill** [comment]
Raymond Hill
2020-02-06

-----

When you force an update of uBO, you need to restart it or restart Firefox, as uBO will not restart immediately upon update to avoid loss of user data. Be sure the version in the popup panel is the latest dev build one. Sometimes I found that I need to disable uBO, force an update, quit Firefox, launch Firefox then re-enable uBO.

**Comment by gorhill** [comment]
Bram Pitoyo
2020-02-18

-----

Hi Raymond,

On the last dev build, I wasn’t able to follow your instruction as the “uiFlavour” pref wasn’t there to modify. I was finally able to download the latest build and test the changes today.

I was really impressed by all the changes you’ve made. Thank you very much for rebuilding your UI and restyling it to fit! It has paid off. And when Firefox Preview releases to the public, uBlock Origin will be the first ever add-on to have a mobile-optimised interface.

At this point, I would consider your work complete. The only other set changes I would make, if you are open to it, are minor. They’re to bring your colour scheme in line with Fenix’s (so you look native), and to bring some UI elements in line with how Android works.

This work is totally optional. If you think that uBO’s desktop branding is important to carry to mobile, you can keep your existing stylesheet. The biggest chunk of work was already accomplished. The mobile UI you’ve built is enough to make uBO feel at home on smartphones.

Here are the optional changes I would propose for your consideration:

1. Change body text colour to match Fenix Text colour

body {
color: black; → #20123a; (want something darker? Try #1d1133)
}


2. Change switch colour to match Fenix Blue colour palette

#switch .fa-icon {
fill: #0060df; (we also have these shades of blue: #054096, #0250BB, #0090ED, #00B3F4)
}


3. Change tool text and fill colours to match Fenix On-Light colour (identical to text)

.tool {
color: #444; → #20123a; (darker shade: #1d1133)
fill: #444; → #20123a; (darker shade: #1d1133)
}


4. Move icon badges from bottom-right to top-right (matches how badges work in Android)

.fa-icon-badge {
bottom: -20%; → top: -20%
}


5. Change hr border colour to match Fenix Separator colour

hr {
border-top: #ddd; → rgba(21,20,26,0.12); (flattened colour: #e3e2e3)
}


6. Change table background colour to match Fenix Grey (Light UI) colour palette

#firewallContainer > div > span {
background-color: #e6e6e6; → #e0e0e6 (alternatives: #cfcfd8, #f0f0f4, #f9f9fb)
}


Screenshot of my result:

<img width="360" src="https://user-images.githubusercontent.com/585534/81931131-9d5ea200-95b7-11ea-8430-4f32a1a08cb6.png">

We also have a dark theme, which you may be able to target with @media (prefers-color-scheme: dark):

<img width="360" src="https://user-images.githubusercontent.com/585534/81931368-fa5a5800-95b7-11ea-877d-edbe4dec3a0f.png">

Perhaps Philipp can help us confirm whether add-ons can detect Fenix’s theme and show a different set of styles to suit?

Either way, we are good for release as it is.

Thanks so much, again!
Bram

**Comment by gorhill** [comment]
Raymond Hill
2020-02-20

-----

Hi Bram,

I did all the changes you suggested below -- they are parts of the uBO version 1.25.0 which I published yesterday.

Things I didn't do yet:

- Dark theme: I admit I will need guidance for the color scheme.

- The scrollable tabs in the dashboard. I remember you mentioning that Mozilla had code for this, I would like to reuse this code in uBO.

- I wonder: should the tooltips in the popup panel be re-designed as well?

**Comment by gorhill** [comment]
Bram Pitoyo
2020-02-25

-----

Hi Raymond,

> Dark theme: I admit I will need guidance for the color scheme.


Good news: I was able to confirm that it’s possible to detect Fenix dark theme through the use of CSS prefers-color-scheme.

I don’t have a uBlock-specific dark theme CSS that I can share. However, in the past, I’ve mocked up what our dark theme would look like on Material Web Components:

https://codepen.io/brampitoyo/pen/Jjoordg

I believe that we can do the same CSS targeting technique for uBO, and achieve a good result.


> The scrollable tabs in the dashboard. I remember you mentioning that Mozilla had code for this, I would like to reuse this code in uBO.


We don’t have a code for this, but would recommend using Material Web Components because we know for sure that it’s going to look good on mobile devices.

Here’s an example of your dashboard layout, that I’ve worked on:
https://codepen.io/brampitoyo/pen/zYGNjwG

Unfortunately, because this involves a third-party design library, I don’t know whether it’s going to be easy for you to integrate into uBO.


> I wonder: should the tooltips in the popup panel be re-designed as well?

Yes, if you’re interested, then I think we should redesign it.

Because there’s no concept of “hover” on mobile devices – and because long-press is commonly reserved for context-menu – we don’t have any of tooltips on Fenix.

Instead, we rely on the icon to be self-explanatory, or for those icons to have clear labels.

For example, we can prefix the words “Cosmetic filtering” or “Remote fonts” with words that conveys the current state of the feature.

For example:

- Cosmetic filtering allowed
- Remote fonts allowed

Alternatively, we could a toggle control that makes it clear whether that feature is ON or OFF. Another alternative is to have checkboxes that the user can check/uncheck.

But in this case, I realise that space is limited. So I’d recommend just making the caption clearer by explaining its current state. Is it on or off? Enabled or disabled? Write that below the icon.

When you tap those icons/captions, firstly we change the icon. If something is crossed, we un-cross it.

Secondly, we change the caption to suit the current state:

- Cosmetic filtering allowed → Cosmetic filtering blocked
- Remote fonts allowed → Remote fonts blocked

We use “allow” and “block” consistently throughout Firefox and Fenix for permissions. But if you like, you can use “On” or “Off” for your features. Or even “Enabled” and “Disabled”. As long as it’s consistent, I think your users will find them clear.

In summary, what I might do on mobile, is to clarify the caption text underneath each icon. On desktop, tooltip might be appropriate as those devices have a mouse-hover state.

What do you think?

Talk again soon,
Bram

**Comment by gorhill** [comment]
Raymond Hill
2020-04-08

-----

Hi Bram,

I finally found some time to work on the dashboard and the latest dev build [1] contains the changes to implement your sketch at [2]

I tried to replicate as well as possible what I could see in your sketch.

I decided that the redesigned dashboard will be used for all versions of uBO instead of just for Firefox Preview. So far I have converted the two most-used panes which suffered the most on small display, the Settings and "Filter lists" panes.

I will be happy to make changes and fine-tune as you see fit. I didn't test the re-design on Firefox Preview because I can't easily install uBO's dev build on it, I used the Responsive Design Mode on Firefox Nightly and I also tried it on main Firefox for Android version.

[1] https://github.com/gorhill/uBlock/releases

[2]
https://app.abstract.com/share/54f1465e-8d1c-4e7b-9ae4-f37e21bef7fe?collectionId=72b94e4e-2e7d-4e43-b5fc-aadd451939ea&collectionLayerId=13e3ba28-ee47-46e8-98f5-6f00948c5168&sha=95d331bab08df7e733f2aa54eb60fc5acff40765

**Comment by gorhill** [comment]
Bram Pitoyo
2020-04-09

-----

Hi Raymond,

Sending you all the screenshots from my test device, below.

<img width="360" src="https://user-images.githubusercontent.com/585534/81931649-70f75580-95b8-11ea-8a96-254937d546a8.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81931789-a439e480-95b8-11ea-81fb-47eaa9d9738e.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81931851-bca9ff00-95b8-11ea-8a66-07854c0796fa.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81931904-cdf30b80-95b8-11ea-9962-6c05e2d59836.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81931976-efec8e00-95b8-11ea-9407-98e6b4060a8d.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81932029-0a266c00-95b9-11ea-9ade-3f42b04e06e8.png">

I really appreciate you for working on this “last mile” feature – pages that many users don’t see, but will make all the difference for those who use them.

Seeing your designs was encouraging. I honestly think that there won’t be much more to do other than the usual suspects: matching font/form element styles, sizes, colours and paddings to Fenix style, making sure that everything is sized properly for touch targets (48dp height if possible), etc. And that’s because you’ve done the hardest part of the work: making the navigation tabs mobile-friendly and horizontally scrollable.

I’m about to go on holiday starting tomorrow until the end of next week. I’ll review the design and send along suggestions when I’m back.

uBlock Origin is still the most Fenix-native looking add-on so far. And with this Settings design, I think it will continue to widen the lead. Congratulations again, and thanks for working on this!

–Bram

**Comment by gorhill** [comment]
Raymond Hill
2020-04-09

-----

Hi Bram,

Thanks for the feedback screenshots, they made me realize the results is not what I thought it would be. I did follow your guidelines regarding color and font size, but I see issues in the screenshots:

- The non active tab buttons in the dashboard are blueish in your screenshot -- I did not expect this. I found this is caused by the fact that these buttons are `a` tags and thus in your case they are rendered as unvisited links. I fixed this by converting these elements to `span` tags.

- I realize the font size is not as `16px/24px` as you suggested. I found this was caused by a bad cut & paste of CSS code, which caused the desktop font size (14px/21px) to be used. When I used the Responsive Design Mode in Firefox's dev tools the font appeared larger and I wrongly assumed that the `16px/24px` was really being used.

The fixes above are in latest dev build of uBO[1].

Hmm, as I write this, I realize that by converting the tab buttons to `span` tags, I may have made uBO's dashboard less accessibility-friendly -- the buttons can't be selected using the keyboard, I will look into this.

[1] https://github.com/gorhill/uBlock/releases

**Comment by gorhill** [comment]
Bram Pitoyo
2020-04-19

-----

Hi Raymond,

Here are screenshots from the latest dev build (1.26.3b0):

<img width="360" src="https://user-images.githubusercontent.com/585534/81932213-6e493000-95b9-11ea-9755-d067a716d58b.png">


<img width="360" src="https://user-images.githubusercontent.com/585534/81932226-71dcb700-95b9-11ea-9d7c-5ef22d96a701.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81932316-9d5fa180-95b9-11ea-9c7d-3a7399f9fa8e.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81932322-a05a9200-95b9-11ea-803c-f1f3bd3485f6.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81932386-bd8f6080-95b9-11ea-90bd-e5139f1e1fc8.png">

<img width="360" src="https://user-images.githubusercontent.com/585534/81932401-c08a5100-95b9-11ea-8d4f-07901402ac82.png">

Responding to the issues you’ve written:

> The non active tab buttons in the dashboard are blueish in your screenshot

This seems to have been fixed. The non-active tab colours are now dark grey, just like normal text.

> I realize the font size is not as `16px/24px`

Although there’s no apparent font size change on the last set of screenshots and these ones, I can confirm that the size you’ve specified matches ones found in our Main menu and Settings. In other words, they’re 16px. Compare the font size of your tab titles with the screenshot below:

<img width="360" src="https://user-images.githubusercontent.com/585534/81932481-e283d380-95b9-11ea-9f9f-3df3a5462407.png">

On the contrary, a 14px text looks smaller, like what we use under Settings → Add-ons → Details:

<img width="360" src="https://user-images.githubusercontent.com/585534/81932577-021afc00-95ba-11ea-87bb-dd60a1b0f5fd.png">

Lastly, here are a couple of small visual recommendations that I’ve promised to send:

1. Have you thought about changing the blue highlight colour of your checkboxes, into the same accent colour that we use (#312a65)? Alternatively, you can use a brighter accent colour, which is #592acb. Feel free to pick!

<img width="360" src="https://user-images.githubusercontent.com/585534/81932656-1c54da00-95ba-11ea-8b59-13c523bec56b.png">

2. Have you thought about changing the colour of your links, into the same colour that you use for active tabs (same colour that we use for links – #592acb)?

<img width="360" src="https://user-images.githubusercontent.com/585534/81932725-3393c780-95ba-11ea-8900-dcc2571602bc.png">

3. On your dashboard Settings sub-page, I notice that you have a few headings: “Privacy” and “Default behaviour”. And on your Filter lists sub-page, you have various headings like “Built-in”, “Ads”, “Privacy”, “Annoyances”, etc.

If you’re wondering about how we style them, you can look at our Settings for inspiration. We also have a few headings: “General”, “Privacy and security”, “About”, etc.

Each heading is set like this:

- font-family: Metropolis
- font-weight: 600
- font-size: 14px
- color: #312a65
- horizontal ruler colour: #e0e0e6

<img width="360" src="https://user-images.githubusercontent.com/585534/81932810-5920d100-95ba-11ea-8730-f19a9b1b93b2.png">

4. For all of your list items, have you thought about making their measurements, width/height and paddings consistent? See this example:

<img width="360" src="https://user-images.githubusercontent.com/585534/81932872-72298200-95ba-11ea-9803-6811dba9c831.png">

5. Have you thought of matching your button measurements and style to ours?

- font-family: Metropolis
- font-weight: 600
- font-size: 14px
- color: #312a65
- background: #e0e0e6
- height: 36dp
- left/right padding: 16dp

We follow Android Material button specs closely: https://material.io/components/buttons#specs

Your buttons are on the left, and Fenix buttons are on the right:

![a](https://user-images.githubusercontent.com/585534/81933094-d0566500-95ba-11ea-92b0-a7a4d446647b.png)

6. On the Filter lists sub-page, you have the +/- symbol to indicate that each heading can be expanded. In Android, we use the downwards arrow for this purpose.

Because you aim to make your mobile and desktop UI consistent to some degree, I’m not sure if a downward arrow would be appropriate to use in this case? It’s just something I put here to consider.

Your UI is to the left, and Android System Settings UI is to the right:

![a](https://user-images.githubusercontent.com/585534/81933145-e532f880-95ba-11ea-90ba-a6df2e42b7cb.png)

You’ll notice that all these recommendations are just small refinements on top of all the hard work you’ve already done. I hope I’ve put them in a format that makes it easy for you to consider, in case you’re interested!

Talk soon,
Bram

**Comment by gorhill** [comment]
Raymond Hill
2020-05-13

-----

Hi Bram,

I was wondering if you mind that I create a new issue on uBO's repo to reflect the relevant email exchanges we had since the beginning?

The main idea is that your contributions to uBO are made public and in general I just like that uBO is development details are completely open to the public. There are also many details in your past emails which I know I still need to address so having those details collected in an opened issue will document what is still left to do.

I have no problem if you prefer to decline so don't feel like you have to accept.

**Comment by gorhill** [comment]
Bram Pitoyo
2020-05-13

-----

Hi Raymond,

I don’t mind at all! In fact, I was emailing you initially just in case you’d prefer mobile contributions to be made private until you’re ready to announce them publicly.

In the future, it’s good to know that I have your permission to directly file GitHub issue and get community’s eyes on it, instead of working privately all the time :-)

Thanks for being a great collaborator. I look forward to seeing the issue in the repo.

Bram

**Comment by gorhill** [comment]
Related issue: https://github.com/uBlockOrigin/uBlock-issues/issues/1005

**Comment by gorhill** [comment]
Just to clarify some choices that were made in the new popup panel, there are changes that I have made on my own and for those who do not like them, I am the one to whom you should complain. Here are the choices I made on my own:

Bring the per-site switches close to the power button: The per-site switches affect the current site, and may requires a reload of the page, or may requires clicking the padlock or eraser, hence why it makes sense to have them closer to the power button at the top. The only reason they were at the bottom in the previous panel is just because I originally threw them there _years ago_ to not change what was above, there was no profound reasons to have them there.

Move the less often used "go somewhere" icons toward at the bottom: Because they are less often used, and well separated enough from the per-site switches such that they are not to be confused as per-site switches.

I added the hostname label of the current site. This is something I have tried numerous times to accomplish with the old panel, be that did not work well (panel was too narrow). It finally fit in there well with the new panel.

I extended the existing "More" button into More/Less buttons to give more control about which sections are visible in the popup panel. This takes care of many issues including those which were not made official in the issue tracker. For example there had been requests in the past about not showing the "Blocked since install" count.

**Comment by RUSshy** [comment]
I don't think it's a good idea to listen to Mozilla's advice, they aren't good at UI/UX, just look at firefox.. they lost all their users in the past years, and UI/UX is one of the reason of the exodus

![image](https://user-images.githubusercontent.com/18348637/82113763-cff3d080-9758-11ea-97d1-9760d689c45e.png)


I vote for sticking to the original design, maybe just add a dark theme and make some controls more accessible for people with disabilities


> Have you thought of matching your button measurements and style to ours?

Haha, i see what's their goal there, they want to confuse people into thinking it's the same people/company/sect ;)

**Comment by guillotjulien** [comment]
> I don't think it's a good idea to listen to Mozilla's advice, they aren't good at UI/UX, just look at firefox.. they lost all their users in the past years, and UI/UX is one of the reason of the exodus

Maybe you need to take a closer look to Firefox preview, firefox focus and the new desktop interface then. They changed a lot of things for the best.

Design choices are at the end only a matter of personal tastes, for exemple, I feel that Chrome UI sucks. But the advices provided here (if you exclude the part with matching colors), are advices that make sense for most mobile interfaces. Bigger controls and text are easier to interact with on mobile for most users.

**Comment by andyljones** [comment]
[This thread has made it to the front page of HN](https://news.ycombinator.com/item?id=23200291). You may want to lock the thread for a while, and delete this comment either way.

**Comment by objection-1** [comment]
@RUSshy 

1. You claim an "exodus" but show relative data (%). The percentage can decrease by simply increasing the total amount of users. And into the reference time falls the rise of mobile and tablet with the increase of usage in Chrome and Safari. So, please use *different* data to prove an *exodus*.
2. You claim Firefox has bad UI/UX yet argument against one of the most popular extensions using the same UI than the main program. Isn't this a point that you should consider?
3. There is so much more to consider (e.g. Google.com and youtube.com encouraging users to switch to Chrome?) than just claiming people are leaving with bad UI design being "one of the reasons".

**Comment by hjek** [comment]
New design is lovely. Minor new issue: The pop-up is quickly resized four times before it displays (causing bright flashes if a dark browser theme is used).

**Comment by RUSshy** [comment]
also: sneakingly asking author of open source project to change UI to match 3rd-party identity by email, skipping any open and public discussion about it

wonder what other sneaky things Mozilla is doing ;)

**Comment by gorhill** [comment]
> props and respect to gorhill for being open and transparent!

Please read carefully:

> In fact, I was emailing you initially just in case you’d prefer mobile contributions to be made private until you’re ready to announce them publicly.

It's me who chose to keep working on this through email, I didn't want everybody's subjective opinion to get in the way of work done.

Now, I am not interested in feedback rooted in grudge -- this is not feedback which is actionable. I just want to fix things that are not working properly on the desktop version of the popup panel, and this can be done only when people submit specific, reproducible issues.

It turns out that by finally releasing 1.27.0 I received more feedback about glitches than I did with the dev build -- I have fixed those in [1.27.2](https://github.com/gorhill/uBlock/releases/tag/1.27.2) which I have submitted yesterday to AMO. There are a few more issues being raised currently, I will work on these for probably another quick 1.27.4 release.

@hjek What is your browser/version? When you say _"bright flashes if a dark browser theme is used"_, do you mean in the popup panel? The popup panel does not yet support a dark theme, so I would like more details as to what you are describing. I am also not sure to understand _"pop-up is quickly resized four times before it displays"_, since the popup is set to become visible only after the first time it's populated with data related to the current site.

**Comment by gwarser** [comment]
>I am also not sure to understand "pop-up is quickly resized four times before it displays",

I sometimes (when first time opened?) see blank popup panel being 250px wider than in most expanded state (all sections visible) and then shrinks to normal size. It's like two or three frames of animation. I wonder if prefers-reduced-motion is affecting this.

**Comment by gorhill** [comment]
> when first time opened?

Browser/version? Whichever browser it happens on, does it also happen in a different browser? When you say _"first time opened?"_, do you mean this happens once only and then it no longer occurs until the browser or uBO is restarted?

**Comment by filips123** [comment]
I have some feedback about new UI design:

* When blocked domains are displayed on left side of popup, darker background of their section ends immediately after list ends:

  ![blocker-domains-background](https://user-images.githubusercontent.com/16626308/82119697-f7f72a00-9780-11ea-9c1b-5e79eb18d9c0.png)

  I think that background should always end at the bottom of popup regardless of list length.

* When mouse hovers on/off button, there is some grey background behind it which ends right where button ends and looks somehow "cropped":

  ![button-background](https://user-images.githubusercontent.com/16626308/82119734-3ab90200-9781-11ea-998b-40242f053d96.png)

  I think that it would be better if background on hover won't be displayed at all, like it was on old UI.

* Some (probably new) buttons are not translated yet.

* Otherwise new UI looks good.

**Comment by ericschdt** [comment]
> > I am also not sure to understand "pop-up is quickly resized four times before it displays",
> 
> I sometimes (when first time opened?) see blank popup panel being 250px wider and half the height than in most expanded state (all sections visible) and then shrinks to normal size. It's like two or three frames of animation total. I wonder if prefers-reduced-motion is affecting this.


> Browser/version? Whichever browser it happens on, does it also happen in a different browser? When you say _"first time opened?"_, do you mean this happens once only and then it no longer occurs until the browser or uBO is restarted?

For what it's worth, I see it too: 
> Firefox 76.0.1, uBO 1.27.0 on Arch KDE
For me it happens practically always. I will test it in other browsers later.

**Comment by gorhill** [comment]
@filips123 

> Some (probably new) buttons and texts are not translated yet.

This has to be handled at <https://crowdin.com/translate/ublock/13/enca-sl> -- I just pull all translation work at regular interval; all translation work is done at Crowdin.

I will keep the other feedback in mind for a next normal release.

**Comment by hjek** [comment]
> @hjek What is your browser/version?

Firefox 75.0 (64-bit) on Fedora

> When you say "bright flashes if a dark browser theme is used", do you mean in the popup panel?

Sorry for lack of details. I'm using the [Dark Reader add-on](https://darkreader.org/) and enabled the built-in dark Firefox theme. When I click on the uBlock origin add-on icon to display the extension pop-up, at first this pop-up is a blank, white rectangle which covers most of the browser window for a few milliseconds. It then changes to a different sized rectangle for a few more ms, and does so about 4 or 5 times before displaying the actual pop-up content (with the large blue **on / off** button in it). This didn't happen before the UI redesign.

I find it a bit hard on my eyes to have sudden bright flashes on an otherwise dark screen.

My guess is that the pop-up is populated with elements first and then afterwards some elements are hidden. I am also using a fairly low-spec laptop which may make the issue more noticeable.

Thank you so much for your work on this add-on. It is much appreciated.

**Comment by gwarser** [comment]
> Browser/version? Whichever browser it happens on, does it also happen in a different browser? 

Firefox nightly mostly, but on Chrome it's also visible - just faster.

>When you say _"first time opened?"_, do you mean this happens once only and then it no longer occurs until the browser or uBO is restarted?

I mean it happens once, when I need to access panel. After that it's not reproducible no mater how many times I try. But after few minutes when I need panel again I see it again.

**Comment by uselessgithub** [comment]
Great upgrade in the UI @gorhill ! It looks modern and clean, but without overdoing it.
I noticed a tiny visual glitch (v 1.27.0 on Firefox 77.0b6), when I click "More"/"Less" buttons, the icons in the popup move around a bit (like a few pixels to the side then back to original position).

**Comment by gorhill** [comment]
I made some changes, available in [latest dev build](https://github.com/gorhill/uBlock/releases). These changes are mostly guesses since I can't reproduce or see those glitches on my side:

- Flashes of white backgrounds when the popup opens in dark colors mode: I am taking a guess that leaving the background to the default color is what caused this issue.
- Spurious resizing of the popup panel: I am taking a guess that a `width: 100%` is what caused this issue.
- I restored the ability to change the popup font size through advanced setting `popupFontSize`, for those who do not like the default `14px` font size -- until I finally am able to invest time in fixing <https://github.com/uBlockOrigin/uBlock-issues/issues/397>. There is no guarantee offered if the set font size departs too much from the default `14px`. I tried from `11px` to `16px` and all seemed ok.

**Comment by Mouvedia** [comment]
> When mouse hovers on/off button, there is some grey background behind it which ends right where button ends and looks somehow "cropped"
> I think that it would be better if background on hover won't be displayed at all, like it was on old UI.

I agree; only the icon's color should change not its background.
Maybe a drop shadow instead ?

**Comment by LegitLlama** [comment]
Hi,

I am in windowed mode, with uBO's icon on the left side of the address bar. There's quite a lot of screen space to the left of the Firefox window, and uBO's pane tries to open there very quickly (i.e. towards the left) before moving back to the right to then be displayed properly.

My understanding is that there may not be enough space for the final pane dimensions to fit towards the left, to it moves to the right.

Expected behaviour would be for the pane to open over the Firefox window without flashing to the left first. uMatrix has the same issue described here except it flashes for longer.

&nbsp;

It might be more visible to computers with a poor CPU.
uBO 1.27.3b0 (I manually [changed](https://github.com/gorhill/uBlock/releases/download/1.27.3b1/uBlock0_1.27.3b1.firefox.signed.xpi) the download URL to `b1` but once installed it says `b0`)
Firefox 76.0.1 on Windows 7 64-bit

**Comment by gorhill** [comment]
@LegitLlama What you are describing is a browser issue, uBO has no control over the viewport in which it is asked to render. When you click the extension icon, Firefox creates a viewport then load the extension popup HTML document, and resize the viewport according to the size of the HTML document. You need to report what you experience to Firefox devs at <https://bugzilla.mozilla.org/home>.

**Comment by hjek** [comment]
> These changes are mostly guesses since I can't reproduce or see those glitches on my side:

Issue is still there in the latest dev build. Thanks for trying. I should probably try fixing it myself since it happens on my device.

**Comment by LegitLlama** [comment]
@gorhill I tried reproducing it with other add-ons but couldn't. Could uBO and uMatrix have some particularity in their HTML document that facilitates Firefox making a wrong initial assumption ? Well either way thanks, I won't clutter this issue if it's not something that can be helped with the redesign :)

**Comment by gorhill** [comment]
@hjek 

>  should probably try fixing it myself since it happens on my device.

My next guess would be to avoid having to compute the width of the firewall pane. Could you try:

At <https://github.com/gorhill/uBlock/blob/ed49b42959ed8d612e4e82e124ae34309c70fb4e/src/css/popup-fenix.css#L525>, replace `min-width: 360px;` with `max-width: 460px;`

At <https://github.com/gorhill/uBlock/blob/ed49b42959ed8d612e4e82e124ae34309c70fb4e/src/css/popup-fenix.css#L527>, replace `width: min-content;` with `width: 32em;`.

This way both the basic and firewall panes will have now a hard-coded width -- maybe using `min-content;` requires too much complications for Firefox to come up with a final computed width.

**Comment by gorhill** [comment]
There is an unfortunate issue regarding the rendering of the total blocked count. It's meant to render in a compact form according to the current locale. For example, I get the following in Nightly:

    3.989M or 20%

While I get the following in Firefox 76:

    3,989,000 or 20%

Having Firefox stable not render the compact form contribute to the panel being larger than intended.

The MDN documentation states that the [Intl API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/resolvedOptions#Browser_compatibility) is completely supported since Firefox 29, and I relied on this.

**Comment by hjek** [comment]
> My next guess would be to avoid having to compute the width of the firewall pane.

That didn't work.

As a random observation, commenting out the [`display: flex` rule for the `#panes` selector](https://github.com/gorhill/uBlock/blob/ed49b42959ed8d612e4e82e124ae34309c70fb4e/src/css/popup-fenix.css#L30), makes the flickering stops entirely (and breaks the layout).

Not sure if it's helpful but here's a recording:
![flicker](https://user-images.githubusercontent.com/11958965/82130840-4206fc80-97cf-11ea-9664-828e5b3b2acd.gif)

**Comment by gorhill** [comment]
> commenting out the display: flex rule for the #panes selector, stops the flickering entirely (and breaks the layout).

Thanks for the investigation. The old popup panel is also using `flex` for the panes. The difference might be that with the old panel, the panel was rendered for the desktop by default, while the new one is rendered for small displays by default. I will try to revert the new one to be rendered for the desktop by default.

**Comment by jspenguin2017** [comment]
The new popup is definitely better than the old one, but I think the way it expands and contracts is quite counterintuitive. When I click `More`, I expect more stuff to come out from the bottom, but that's not always the case. Since the popup expands and contracts 6 times, I quickly lose track of what appeared or disappeared. 

I usually don't like unnecessary animations, but this might be a case where some animations make sense. I'm thinking maybe the piece that is appearing will change into a different background color then slide or fade in (or slide or fade out for disappearing). This should make it easier to know what the `More` and `Less` buttons just did.

**Comment by Aeriem** [comment]
There's a very minor inconsistency in the Filter lists panel: "Import..." checkbox remains the same as before the redesign.
Reproduced using uBlock Origin 1.27.0 in Firefox 76.0.1 (Windows) and Firefox 68.8.0 (Android).

![Checkbox](https://user-images.githubusercontent.com/45237401/82142840-a1045a00-983f-11ea-8227-67317689837a.png)

**Comment by gorhill** [comment]
> minor inconsistency in the Filter lists panel

Thanks for the feedback, I meant to fix this but I kept forgetting about it. The fix will be in next dev build.

**Comment by Aeriem** [comment]
You're welcome, the redesign looks dashing. 😃
Definitely a great decision to extend it to desktop installations.
Thanks for the quick reply!

**Comment by gorhill** [comment]
Currently the pain point in the popup panel (beside the glitch detailed above which will hopefully be fixed in the next dev build) is the fact that in some locales, the text strings can be quite long, and though this works fine when uBO's panel is in portrait mode (mobile or overflow menu), it's tricky in landscape mode (desktop). Here is an illustration of this, showing various languages, I have vertically offset the count strings from the label strings to avoid text wrapping:

![a](https://user-images.githubusercontent.com/585534/82143757-27a34200-9814-11ea-9e9f-2217cbac1642.png)

I am not sure how to best deal with this:
- Ask the translators to be more concise; or
- Position the text strings differently -- maybe as above but centered in the panel as this would allow reducing the default width of the main panel

**Comment by andrewshadura** [comment]
> Ask the translators to be more concise

As a person who speaks some of the languages above, I’m not sure how those things would be formulated more concisely. It rarely works this way for Russian an German, for example :slightly_smiling_face:

**Comment by gorhill** [comment]
> not sure how those things would be formulated more concisely

Ok so I will not burden translators.

In the end I went for stacking vertically the strings in landscape mode (when the firewall pane is meant to open on the left side). This solves the text strings wrapping due to lack of width ([example](https://abload.de/img/unbenanntouk41.jpg)), and will allow for a less wide basic panel.

@hjek The panel is now in landscape mode by default on desktop, so there should be done big layout modification when the panel is first opened. I hope this fixes the glitchy viewport resize seen in your gif.

**Comment by gorhill** [comment]
@andrewshadura

> not sure how those things would be formulated more concisely

Thanks for the feedback, so I prefer to not burden volunteer translators.

In the end I went for stacking vertically the strings in landscape mode (when the firewall pane is meant to open on the left side). This solves the text strings wrapping due to lack of width ([example](https://abload.de/img/unbenanntouk41.jpg)), and will allow for a less wide basic panel.

@hjek The panel is now in landscape mode by default on desktop, so there should be done big layout modification when the panel is first opened. I hope this fixes the glitchy viewport resize seen in your gif.

**Comment by filips123** [comment]
I don't know if this happens just for me (I'm currently having some problems with Firefox on Linux), but panel sometimes glitches while showing/hiding domain list:

[Screen recording](https://github.com/uBlockOrigin/uBlock-issues/files/4640968/screenrecording.zip)

It's ZIP because I can't upload MP4 to GitHub. I don't know if it is visible on video, but sometimes when clicking on more/less buttons to show/hide domain list, panel will first "try" to show new content and then resize, resulting in some visual glitch.

**Comment by gorhill** [comment]
> panel sometimes glitches while showing/hiding domain list

This has always been the case with Firefox, this is not specific to the new panel.

**Comment by gorhill** [comment]
Probably the last visual fine tuning before releasing 1.27.4: maybe make the More/Less buttons a bit less high, but I am completely undecided what is better, either visuals work for me, so I am going to leave it for vote:

For reference, current visual on the left, proposed visual on the right:

![b](https://user-images.githubusercontent.com/585534/82161024-7a5a1980-9867-11ea-8e01-6f6e3850315d.png) ![a](https://user-images.githubusercontent.com/585534/82161020-77f7bf80-9867-11ea-9892-909e3e3f12e8.png)

Vote about whether you agree with the change: vote :-1: if you prefer left, vote :+1: if you prefer right.

(To be clear this applies only to the desktop version of the panel, the mobile version stays the same as in 1.27.0.)

**Comment by filips123** [comment]
> This has always been the case with Firefox, this is not specific to the new panel -- it has always happened to me too.

Can it be fixed?

Also, are there any changes regarding feedback in my first comment? I added new Slovenian translations so strings are now correctly translated but what about other two points?

**Comment by gorhill** [comment]
> Can it be fixed?

By Firefox devs, the the popup panel is ultimately directly resized by Firefox, not by uBO.

> what about other two points?

Sorry, you lost me. What other two points?

**Comment by gwarser** [comment]
https://github.com/uBlockOrigin/uBlock-issues/issues/1027#issuecomment-629638890

**Comment by FatOrangutan** [comment]
>maybe make the More/Less buttons a bit less high, but I am completely undecided what is better, either visuals work for me

Considering that making uBO look more at home in Firefox seems to be one of the goals behind this redesign, perhaps make the two buttons look like their closest equivalent in the Firefox UI, like maybe the two buttons at the bottom of the 'Bookmark this page' panel:

<details>
<summary>Click to expand</summary>

![Capture](https://user-images.githubusercontent.com/8336638/82162047-38f55880-98bf-11ea-9282-b701b38f3084.PNG)

</details>

...but without styling either of the buttons as the default one, i.e., both `More` and `Less` buttons would look like the `Cancel` button in the screenshot above.

Which, practically, just means that `#moreOrLess` gets a `column-gap` of 0.9em, which is pretty close to what you are currently using right now: 0.875em.

**Comment by SeanDS** [comment]
> ![b](https://user-images.githubusercontent.com/585534/82161024-7a5a1980-9867-11ea-8e01-6f6e3850315d.png) ![a](https://user-images.githubusercontent.com/585534/82161020-77f7bf80-9867-11ea-9892-909e3e3f12e8.png)

While we're on the topic of that screen, would it not be better to say "Blocked on this page... 8 (19%)" and not "Blocked on this page... 8 or 19%" (also for blocked since install etc.)?

**Comment by gwarser** [comment]
I translated it like this in Polish.

**Comment by gwarser** [comment]
http://www.abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk.com/

![image](https://user-images.githubusercontent.com/886325/82204973-189cbc80-9906-11ea-9f95-b799ec600806.png)


- long domains can make popup even wider
- there is a gap between tools and switches with `popupPanelDisabledSections 1`

**Comment by uBlock-user** [comment]
> long domains can make popup even wider

You can reproduce this in Extension page settings too, by visiting dashboard, due to extensively long extension id generated by Firefox.

**Comment by gorhill** [comment]
So given the vote count as of writing, 9 vs. 2, the more/less button will be less high in next dev build.

@filips123 open an issue for the two other points, this ensures I will not forget about these. However, for the first point, it's unlikely I will agree after I thought more about it. In my opinion empty rows will just be confusing, as if the extension broke and forgot to fill in these rows. It would also be difficult fo ruBO to know how much empty vertical space needs to be filled. The second point I find is valid, I did spend quite an amount of time on that one, including making use of drop shadow through the CSS `filter` property but in the end I was unable to make my mind and decided to just stick to whatever is in use in the rest of the popup for the time being.

@FatOrangutan Those Firefox buttons are meant to be temporary, until the dialog closes. In uBO they are always present, they are meant to configure the panel to one's liking and then will probably be rarely used afterward.

@SeanDS The change will be in the next dev build.

@gwarser Fixes will be in the next dev build.

**Comment by Youri111** [comment]
1px is missing when hovering zapper,picker, logger or dashboard item (at bottom).
The first row of tools (media, fonts, ...) has not that issue. This looks much nicer.
![ubo](https://user-images.githubusercontent.com/65547477/82217771-345e8d80-991b-11ea-9d18-79670ca5599f.png)

**Comment by filips123** [comment]
I created #1048 and #1049 for my suggestions.

**Comment by gorhill** [comment]
> 1px is missing when hovering zapper,picker [...]

I can reproduce this. I think this happens because of fractional-pixel positions/dimensions. I am not sure how this can be fixed. For sure this is not trivially fixed, so best to open an issue about this or else I will just forget about it.

**Comment by Aeriem** [comment]
Some observations in uBO 1.27.2 (Firefox 76.0.1):

1. In the Settings pannel, buttons are aligned to the right of the screen.

![Buttons on the right](https://user-images.githubusercontent.com/45237401/82302428-df258900-99b9-11ea-8087-e3b36ce91547.png)

2. Element picker mode features pre-1.27.0 buttons.

![Element picker](https://user-images.githubusercontent.com/45237401/82303689-966ecf80-99bb-11ea-94cd-d40321745c01.png)

3. @brampitoyo suggested shortening some strings in [#1027 (comment)](https://github.com/uBlockOrigin/uBlock-issues/issues/1027#issuecomment-628602103). I did not catch your answer to this. On smartphones, it could be a way to reclaim some space, especially in some locales.

**Comment by uBlock-user** [comment]
> In the Settings pannel, buttons are aligned to the right of the screen.

Addressed -- Reverted back to left in the dev build.

**Comment by Aeriem** [comment]
> Addressed -- Reverted back to left in the dev build.

Thanks for letting me know. 😃

**Comment by ghajini** [comment]
@gorhill have you removed tooltips feature on popup panel (normal user want to know what buttons meant for?)
![IMG_20200519_233043](https://user-images.githubusercontent.com/20338483/82361948-34e54a00-9a29-11ea-975a-9bba8168395b.jpg)

**Comment by gwarser** [comment]
@ghajini 
![image](https://user-images.githubusercontent.com/886325/82377362-d79aca00-9a23-11ea-846b-b2aca8c30311.png)
![Screenshot_20200519-230434](https://user-images.githubusercontent.com/886325/82378321-357be180-9a25-11ea-8943-e75c8bfbe671.png)

**Comment by minig0d** [comment]
Sorry I know my 2 cents is a bit late in coming to the party but have some suggestions that I think are still worth mentioning FWIW.  If any of the below doesn't make sense and you're still open to the suggestions, happy to do a mockup or wireframe...

- In the Advanced part of the Overview section:
	- Add a column header with either the text or icons to remind which column is global vs. local
	- Rather than just one long grid with those two columns, consider horizontally splitting the chart into 3 sections:
		1) The global things (i.e. all (and images?))
		2) 1st party (and have inline scripts, 1-st party scripts, and then the actual domain/subdomain which is currently under the 3rd party frames) (i.e. cluster all these together since they are all logically together)
		3) 3rd party (and include "3rd-party, "3rd party scripts", "3rd-party frames", followed by all the other domains)

- Back in the "basic" section:
	- Consider a "rebrand" of the element zapper mode vs. picker mode to more clearly differentiate the purpose. Ex. rather than "Enter element zapper mode" and "Enter element picker mode" something like "Temporarily zap elements from page" and "Select elements to permanently filter"
	- Regarding the quick toggles for cosmetic, fonts, large media, js, I would suggest removing the dividing line so they are in the same visual box as the "master power switch" since that whole new section is now dedicated to toggling.
		- Maybe just personal opinion but in the many years using uBO I've never used these toggles so I suspect others don't either.  Thought about burrying them all behind the master power switch in one of those downward arrow on the right "additional options" dropdowns?  If not behind the master power switch potentially burrying the large media and remote fonts (which I would bet are used significantly less than any of the other buttons) to reduce "clutter"?
	- in the domains connected can remove the word "out" (i.e. 7 of 8 is just as readable and looks better)
	- If the intent is for the DOM inspector to be the primary place for creation of exception rules, suggest adding a direct link from the overview directly to it.  Probably 90% of the time I even touch uBO that is not creating an additional block, is to have to whitelist something (usually due to a filter list), so IMO, there should be quick access to whitelisting.

- Kind of unrelated to the redesign but having to go to the manual page myself in order to write this made made me realize how much benefit could be gained (for both new users and us 5+ year ones) would be implementing those pop up "walk-through" instruction modals to "introduce" people to uBO.  1) A really brief upon installation and 2) more feature specific ones upon accessing a feature for the first time. (ex. when they click on Element Zapper mode the first time show a modal briefly explaining what it does and of applicable keyboard shortcuts (i.e. Shift/Delete/Esc/swipe)


Just my 2 cents :)

**Comment by NoXPhasma** [comment]
I think the new UI is okay, but two things I really don't like are:

- No dark mode and it's now super hard to make your own through chrome css
- Everything is way too big, at least add an option to get the old size back

**Comment by uBlock-user** [comment]
> Everything is way too big, at least add an option to get the old size back

Already exists, set `popupFontSize` to `13px` or lower.

> No dark mode 

WIP...

**Comment by NoXPhasma** [comment]
> Already exists, set popupFontSize to 13px or lower.

Where do I find that? I don't see any option in the Popup or the dashboard to set the font size.

> WIP...

Good to hear.

**Comment by uBlock-user** [comment]
In Advanced settings, so become an advanced user vis Settings pane.

**Comment by gwarser** [comment]
>     * In the Advanced part of the Overview section:
>       
>       * Add a column header with either the text or icons to remind which column is global vs. local

Declined many times already

>       * Rather than just one long grid with those two columns, consider horizontally splitting the chart into 3 sections:

This will suggest they are separate, but they depends on each other (rule precedence)

>     * Back in the "basic" section:
>       
>       * Consider a "rebrand" of the element zapper mode vs. picker mode to more clearly differentiate the purpose. Ex. rather than "Enter element zapper mode" and "Enter element picker mode" something like "Temporarily zap elements from page" and "Select elements to permanently filter"

Hmmm...

>         * Maybe just personal opinion but in the many years using uBO I've never used these toggles so I suspect others don't either.

Some use them many times a day

>Thought about burrying them all behind the master power switch in one of those downward arrow on the right "additional options" dropdowns?  If not behind the master power switch potentially burrying the large media and remote fonts (which I would bet are used significantly less than any of the other buttons) to reduce "clutter"?

This is the default for new installations? Click again on "Less"?

>       * If the intent is for the DOM inspector to be the primary place for creation of exception rules, suggest adding a direct link from the overview directly to it.  Probably 90% of the time I even touch uBO that is not creating an additional block, is to have to whitelist something (usually due to a filter list), so IMO, there should be quick access to whitelisting.

This is rather advanced tool, for easy usage there is already cosmetic filtering switch.

**Comment by volcbs** [comment]
@gorhill 
![Annotation 2020-05-21 172811](https://user-images.githubusercontent.com/12575216/82550324-86f0b180-9b88-11ea-8ad8-7bfcfd6bfaf7.png)
Somehow the font didn't change but everything else is shrunk when I changed the "popupFontSize"

I'm using chromium edge (chromium v84) btw.

**Comment by gwarser** [comment]
@volcbs append `px` to the number.

**Comment by volcbs** [comment]
@gwarser neat, also need to type `10px` instead of `10 px` too. thanks anyway

**Comment by zakius** [comment]
I consider these 6 buttons used for domain filtering a bit too tall but definitely not wide enough on mobile

And the overall look of interface and settings is kinda off, looks like mobile app made to *look good*, gives me the same feeling as DevTools in Quantum

**Comment by minig0d** [comment]
> > * In the Advanced part of the Overview section:
> >   
> >   * Add a column header with either the text or icons to remind which column is global vs. local
> 
> Declined many times already

Must be requested multiple times because a lot of people think it's a good idea...  :)
Maybe a watermarked icon as an image background as a compromise?

> 
> > * Rather than just one long grid with those two columns, consider horizontally splitting the chart into 3 sections:
> 
> This will suggest they are separate, but they depends on each other (rule precedence)
> 
None of my suggestions are intending to suggest that there is a PHYSICAL separation in function (except the zapper/picker), they are simply suggesting that some of the layout doesn't really match up with how we subconsciously expect to see it laid out, so it increases the learning curve.  So again not to suggest there is a physical distinction, just saying when I'm trying to mentally organize what blocking I want to implement or what I am comfortable relaxing based on the the site, I'm thinking... 1) This is what I'm ok with relaxing globally  2) then based on the site I trust their first party content X amount  3) but still don't trust the third party content.  (and based on each of those 3, then whether it should be blocked/allowed globally or locally.  But right now it mentally directs you to the global vs. local decision and then lumps everything besides that in one bucket which is backwards.


> > * Back in the "basic" section:
> >   
> >   * Consider a "rebrand" of the element zapper mode vs. picker mode to more clearly differentiate the purpose. Ex. rather than "Enter element zapper mode" and "Enter element picker mode" something like "Temporarily zap elements from page" and "Select elements to permanently filter"
> 
> Hmmm...
> 
> > * Maybe just personal opinion but in the many years using uBO I've never used these toggles so I suspect others don't either.
> 
> Some use them many times a day
 (sorry this comment was just an introduction leading up to the next)

> > Thought about burrying them all behind the master power switch in one of those downward arrow on the right "additional options" dropdowns?  If not behind the master power switch potentially burrying the large media and remote fonts (which I would bet are used significantly less than any of the other buttons) to reduce "clutter"?
> 
> This is the default for new installations? Click again on "Less"?
Wasn't meaning as far as collapsing the advanced options panel, was meaning that the quick toggles for all the smaller options be placed under a "split button" of the main power button. (or 
![Screenshot](https://i.imgur.com/CVqGkDm.png)

Sorry didn't know the official name but one of these: (and if not a split button, maybe put the buttons in a small accordion/collapsible section or in one of those floating modals when you hover over something... )
![Screenshot](https://i.imgur.com/CVqGkDm.png)

If not doing it on the main power button, possibly collapsing "fonts" and large objects under a split button.  Not earth shattering important but theres just tons of buttons everywhere and there are techniques to help bring order to them and present them in a little bit cleaner of a manner.just a lot of "clutter"


> > * If the intent is for the DOM inspector to be the primary place for creation of exception rules, suggest adding a direct link from the overview directly to it.  Probably 90% of the time I even touch uBO that is not creating an additional block, is to have to whitelist something (usually due to a filter list), so IMO, there should be quick access to whitelisting.
> 
> This is rather advanced tool, for easy usage there is already cosmetic filtering switch.

Sorry not meaning to spotlight the functionality of the tool, just pointing out the lack of symmetry/consistency in workflow that could seemingly be remedied with a direct icon to the DOM inspector like there is one to the logger.  But that's also an inconsistency in itself... I actually really like the little mini "wizard" to build filters/exceptions within the logger and get disappointed when I'm not able to use it to whitelist cosmetic exceptions.

-----

Again none of these are earth shattering, just slight tweaks that would make the layout more intuitive and cleaner.

**Comment by ghajini** [comment]
@gwarser 
Iam on firefox android 68 and firefox android beta 75.0.0 beta 6

Ublock origin 1.27.6

Iam not seeing tooltips yet

**Comment by gwarser** [comment]
>Iam not seeing tooltips yet

@ghajini no icon captions? "tooltips" are only possible on devices with mouse pointer.

>No popup switch is absent on firefox android stable build

Yes, by design, explained above.

**Comment by ghajini** [comment]
Ubo 1.25.2
Long pressing on switches will show 
tooltips
![IMG_20200523_152734](https://user-images.githubusercontent.com/20338483/82727899-ef679c00-9d0a-11ea-99ba-3dbf2bf25bbd.jpg)

 for touch android devices....these buttons now in 1.27.6 don't describe what they meant for?

**Comment by zakius** [comment]
by the way, the "show more, show less" row shows only one more row for me when unfolded, wouldn't it make more sense to just show that row directly instead? or is something more supposed to show?

**Comment by gwarser** [comment]
> At least like this to be done

@ghajini but captions are already here. How it looks on your phone?


>by the way, the "show more, show less" row shows only one more row for me when unfolded,

@zakius click again.

**Comment by zakius** [comment]
ah okay, so it can be folded further, in that case it makes a bit of sense

**Comment by gorhill** [comment]
@ghajini 

> brampitoyo explained how to implement tooltips feature here

[See](https://github.com/uBlockOrigin/uBlock-issues/issues/1027#issuecomment-628604593):

> There are also many details in your past emails which I know I still need to address so having those details collected in an opened issue will document what is still left to do.

**Comment by ghajini** [comment]
@gwarser automatic update form previous version of ubo crippled some ubo settings and I reinstalled ubo 1.27.6 & it looks like good now...
I read above about block all popups switch removed due to limited horizontal space....can you add horizontal scroll bar to adjust block all popups switch

`https://user-images.githubusercontent.com/20338483/82748400-28614880-9dbf-11ea-827d-69cebeab5b8e.jpg`

@ gorhill thanks I will stay tuned to more updates and firefox preview stable....

**Comment by gorhill** [comment]
@ghajini For now you can add `+no-popups` to undocumented advanced setting `uiPopupConfig` to force the no-popup switch to appear in the popup panel.

**Comment by SumatraPeter** [comment]
@gorhill: How do I get to see the new UI on Edge Chromium without installing the [dev build](https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii) from the Chromium web store?

I have the [stable build](https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm) from the Chrome web store installed in Edge Chromium precisely because it's published directly by you, but that version seems to be stuck at 1.26.0 (21 April 2020). Strangely the [Nik Rolls version](https://microsoftedge.microsoft.com/addons/detail/odfafepnkmbhccpbejgmiehpchacaeak) from Edge Add-ons has been updated to 1.27.4 (19 May 2020). How come?

**Comment by volcbs** [comment]
@gorhill Any chance to bring back ub0 version number to popup again like in the legacy one?

@SumatraPeter Huh? Just install the ms store one, dude.

**Comment by hazarek** [comment]
hover color bug. 
![a](https://user-images.githubusercontent.com/36481442/88116442-3e259b00-cbc1-11ea-8830-0fc35e37e37e.gif)

**Comment by filips123** [comment]
I actually made fix for this (which removes shadow) few months ago but I forgot to make PR. Branch with fix is [here](https://github.com/filips123/uBlock/tree/remove-switch-hover-shadow). But I can't made PR anymore because:

> An owner of this repository has limited the ability to open a pull request to users that have contributed to this repository in the past.

**Comment by gwarser** [comment]
@hazarek all buttons have shadow on hover, why this one should be different?

**Comment by uBlock-user** [comment]
That's not a bug, it's by-design in the new UI.

**Comment by filips123** [comment]
It is different for other buttons:

* Shadows for other buttons have some padding around the icon. This button's shadow doesn't have any padding and stops immediately where icon ends.
* Other buttons are located in some kind of "box" (there are lines at the bottom and top) and there are multiple buttons located next to each other. This button doesn't have any "box" and nearby buttons.

**Comment by hazarek** [comment]
> That's not a bug, it's by-design in the new UI.

I am graphic designer this is not a design. please listen to the opinions of the designers. it can turn into a lighter blue when hover not a strange square.

**Comment by filips123** [comment]
I like your first mockup. Except color should maybe be different to be consistant with other elements.

**Comment by hazarek** [comment]
> I like your first mockup. Except color should maybe be different to be consistant with other elements.

developers need to set a color palette standart for the entire ecosystem. if they want a professional look. They have to do it according to merit.

readings for more information - 
https://material.io/design/color/the-color-system.html
https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/

**Comment by hazarek** [comment]
Old and better icons in the logger window.
![c](https://user-images.githubusercontent.com/36481442/88352694-546b5c80-cd63-11ea-82ba-c7e82437c22a.gif)

**Comment by filips123** [comment]
Well, icons are better in redesign. Shadow on hover also looks good most icons, except main on/off switch because it is not contained in any box. For that icon, I think that it would be more appropriate to change color instad of adding shadow.

I tested few availible var colors and I think `fill: var(--default-surface-hover);` would be most appropriate when switch is turned off on hover and `fill: var(--blue-60);` when it is turned on on hover. It's not very drastic change, but would probably help.

This is how it looks:

![button](https://user-images.githubusercontent.com/16626308/88457541-d05ec500-ce87-11ea-9914-20d2cae044d0.gif)

Maybe colors should be changed a bit to provide some more contrast, but in general I think something like this would look nice.

**Comment by hazarek** [comment]
@filips123 it's good but Remove this borders just get gray.
![88457541-d05ec500-ce87-11ea-9914-20d2cae044d0-72 (sürüklenen)](https://user-images.githubusercontent.com/36481442/88457761-c772f100-ce91-11ea-99bc-3040dbd997bd.jpg)

**Comment by hazarek** [comment]
The ugly yellow refresh button does not fit into the layout, IMO it should be refreshed automatically. **"Automatically refresh"** option should be added to the settings. Browser's refresh button and shortcut already exist.
the following message is displayed if "Automatically refresh" checkbox is not selected, 
**"Refresh for the changes to take effect"**
<img width="264" alt="cap" src="https://user-images.githubusercontent.com/36481442/88461971-e46aec80-ceb0-11ea-930a-272182e5ac73.png">

**Comment by garry-ut99** [comment]
- @dzienisz ( https://github.com/uBlockOrigin/uBlock-issues/issues/3258 )

No need for a duplicate thread, the whole discussion is in the current thread, you can continue here.

> dzienisz: You don't know what to do

Most likely to click one of the buttons....if you don't know what buttons do:
   - hover over a button with a mouse cursor over a second 
   and a tooltip will appear which will instruct what the button is for...
   - if still don't understand - you can take a look at the Quick Guide: 
   https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface
